swarm/api/api.go

```diff
@@ -1019,3 +1020,148 @@ func (a *API) ResolveResourceManifest(ctx context.Context, addr storage.Address)
 
 	return storage.Address(common.FromHex(entry.Hash)), nil
 }
+
+// MimeOctetStream default value of http Content-Type header
+const MimeOctetStream = "application/octet-stream"
+
+// builtinTypesLower stores copy of https://github.com/nginx/nginx/blob/master/conf/mime.types
```

Comment:
I personally feel this `mime.types` file should be vendored (or just copied into a Go file) to make it easy to keep in-sync with the upstream file.

I also think it should be clear why this is here (if I saw this I'd probably just think it was unnecessary because the Go mime package does the right thing in pretty much all cases).

Reply:
By vendoring of mime.types file - you mean write some `go generate` command to parse it and gen code? Same approach as here https://github.com/bradrydzewski/go-mimetype (it based on mailcap instead of nginx, but there is same mime file) ?

Reply:
Something like that, yes, which would make it much easier to keep in sync, and more obvious where the types come from.

Reply:
@lmars do you think the overhead is justified in this case? I mean, the mime types base list doesn't fluctuate [that much](https://github.com/nginx/nginx/commits/master/conf/mime.types).